### PR TITLE
CSV Export of all user data

### LINKED ISSUE #508: CSV export of all user data
Implement a dump of all user data for whatever reason they have (switch to other software, further processing, print out). This shall be done in the CSV format.

Based on [initial comment](https://github.com/ezra-bible-app/ezra-bible-app/discussions/476#discussion-3629087) it can be the following CSV table:

| Bible Book | Chapter | Verse | Tags | Note |
| -------------- | ----------- | -------- | ------ | ------ |
| John | 3 | 16 | God's Love; Eternal life |  |
| Ephesians |  |  |   | Written: \_aprox. \*\*60-63\*\* A.D.\_ |
| Ephesians | 1 | 1 | Will of God; Faithfulness | \_Apostle\_ - messenger, a delegate, commissioner of Christ |

Please note, that:
1. There might be multiple tags per verse.
2. There might be a note for the whole book (e.g. Ephesians).
3.  We store notes internally in Markdown format. So the export/dump will be in this format. Right now entering notes is also in this format, but later we might replace it with a simple WYSIWYG editor.

_Originally posted by @zhuiks in https://github.com/ezra-bible-app/ezra-bible-app/discussions/476#discussioncomment-1496233_

### PR DESCRIPTION
**Pull request by tobias-klein**
This fixes #508.

There is a new export button in the App info dialog under the headline "Special functions".

image

### CODE REVIEW

**Comment by tobias-klein** [comment]
@debritto 
Would you like to test this?

Here is a current download for Windows based on this PR:
https://github.com/ezra-bible-app/ezra-bible-app/suites/4763264486/artifacts/132362958

**Comment by debritto** [comment]
Hi @tobias-klein !
I'm so happy to receive this opportunity. You can count on me. I'll install the Ezra App Betta right now. I'll make some tests and let you know what I think.
Warm regards from Brazil!

**Comment by debritto** [comment]
Hi @tobias-klein !

I have tested the CSV Export function and It's working perfect! I already adapted my App to read the values and insert into my database. I would like to thank you my Brother, this feature is a game changer with the project with my students. May The Lord bless you :)

**Comment by tobias-klein** [comment]
> I have tested the CSV Export function and It's working perfect! I already adapted my App to read the values and insert into my database. I would like to thank you my Brother, this feature is a game changer with the project with my students. May The Lord bless you :)

Excellent! Glad to hear that, Christian!
Then this will become available in 1.5.0 (February 19th).

**Comment by tobias-klein** [comment]
@zhuiks I'll merge this once you approve this PR.

File: app/frontend/components/info_popup.js
Comment (zhuiks) [inline app/frontend/components/info_popup.js:228]: Can we do it with vanilla JS, without jQuery?
```javascript
document.querySelector('#export-user-data-button').addEventListener('click', async () => {
  ...
});
```

File: app/frontend/components/info_popup.js
Comment (zhuiks) [inline app/frontend/components/info_popup.js:235]: The time can be increased. Maybe 3s?

File: app/frontend/components/info_popup.js
Comment (zhuiks) [inline app/frontend/components/info_popup.js:236]: Maybe the whole animation can be implemented with CSS animation triggered by applying certain class to it from here. 
But I need to think a little about it.

**Review by zhuiks — DISMISSED** [review]
Everything is great!
I added some comments about using jQuery, but the whole `info_popup` needs to be refactored to the several web components eventualy. So I approve this PR.

File: app/frontend/components/info_popup.js
Comment (tobias-klein) [reply app/frontend/components/info_popup.js:235]: Ok!

File: app/frontend/components/info_popup.js
Comment (tobias-klein) [reply app/frontend/components/info_popup.js:236]: Something for the future then?!

File: app/frontend/components/info_popup.js
Comment (tobias-klein) [reply app/frontend/components/info_popup.js:228]: Ok!

**Comment by tobias-klein** [comment]
> Everything is great! I added some comments about using jQuery, but the whole `info_popup` needs to be refactored to the several web components eventualy. So I approve this PR.

Thanks for reviewing @zhuiks! :)

**Pull request merged**
